### ISSUE TITLE
batch expiry job doesn't report delete marker metrics in batch-status

### ISSUE THREAD

**Issue by krisis** [minio/minio#20506]
## NOTE
If this case is urgent, please subscribe to [Subnet](https://min.io/pricing) so that our 24/7 support team may help you faster.

batch-expiry jobs don't report delete marker status in batch-status. The metrics are calculated but aren't available via exported types in `madmin-go`'s [`ExpirationInfo`](https://github.com/minio/madmin-go/blob/main/metrics.go#L509).

## Expected Behavior

batch-status output for batch-expiry job should include delete marker stats when applicable.
## Current Behavior

batch-expiry jobs don't report delete marker status in batch-status. 
## Steps to Reproduce (for bugs)

1. Use the following yaml to start a batch expiry job
```
expire:
  apiVersion: v1
  bucket: mybucket  # Bucket where this job will expire matching objects from
  rules:
    - type: deleted # objects with delete marker as their latest version
      name: "*" # match object names that satisfy the wildcard expression.
      olderThan: 1h

  retry:
    attempts: 3 # number of retries for the job before giving up
    delay: 500ms # least amount of delay between each retry
```
2. For the above yaml to work, you need the following
- `mc mb ALIAS/mybucket`
- `mc version enable ALIAS/mybucket`
- Upload a few objects and delete some of them; To create delete markers
3. Now run, `mc batch start ALIAS /path/to/expire.yaml`
4. `mc batch status ALIAS $JOBID`
This will not display any delete marker stats.

## Regression

No

## Your Environment

* Version used (`minio --version`):
* Server setup and configuration:
* Operating System and version (`uname -a`):

**Comment by dhananjaykrutika** [comment]
Hi @krisis
I was exploring the batch-start/status commands and also playing around with a few sample jobs.
And it appears, at the time of running `mc batch start`, only objects whose delete markers are older than the specified `olderThan` duration will be picked up for deletion. And status reports the same info correctly.
W.r.t the sample batch expiry job you provided, if `mc batch start` was executed before the 1hr mark, then status will report 0 objects as deleted.
My sense is that this is the expected behavior. Let me know if I'm missing something here.

**Comment by krisis** [comment]
It is possible that the steps to reproduce doesn't demonstrate this issue. I chanced up on it while looking at code.
TL;DR MinIO server records more batch job specific metrics than it exposes to the admin API client (`mc` in this case).

BatchJobStatus handler invokes `batchJobInfo.metric` method to gather job specific metrics. In that method, we can see that each job type have their specific `madmin` exported metric types. `madmin.ExpirationInfo` doesn't have fields corresponding to `batchJobInfo.DeleteMarkers` and `batchJobInfo.DeleteMarkersFailed`. This makes it impossible for admin-API clients to fetch these metrics.

**Comment by jiuker** [comment]
This value also needs to be calculated correctly
